### PR TITLE
add loading logs while fetching repositories

### LINKED ISSUE #80: Loading/progress bar
When you execute the script (`bin/run`), it takes some time to display the result due, depending on the number of repositories. I think it's a good idea to add a progress bar otherwise, it looks like the terminal is stuck.

### PR DESCRIPTION
**Pull request by abdumamdouh**
Fixes #80

### CODE REVIEW

File: src/getRepositories.js [hunk 1]
```diff
@@ -83,8 +83,9 @@ function generateQuery(endCursor, { f }) {
 
 module.exports = async function getRepositories(flags, filter) {
 	// Get all repositories
+  console.log('Loading repositories...');
 	const { points, repositories } = await getRepos(generateQuery, flags, filter);
-
+  console.log('Loaded');
```
Comment (ljharb) [inline src/getRepositories.js:88]: this seems a bit useless, except to "close out" the loading statement, since the table is rendering.

**Review by ljharb — CHANGES_REQUESTED** [review]
I think i'd really prefer to look into some kind of approach where we print to the console, but overwrite it - using process.stdout.write, perhaps?

Either way, we shouldn't be logging inside these reusable functions, and definitely not inside `getRepositories`

File: src/getRepositories.js (hunk 1 above)
Comment (KhatiaIvanova) [reply src/getRepositories.js:88]: @abdumamdouh remove this line

**Comment by ljharb** [comment]
The analogy i have in my head is `printf` vs `echo`. On a shell, you can use `printf` to print characters on a line without moving to the next line, which means you can rewrite that line later.

**Comment by abdumamdouh** [comment]
@ljharb we have used `stdout` instead of `console log`

i think it's ready now to be merged

**Comment by KhatiaIvanova** [comment]
@ljharb Do we need to update something more?

**Comment by ljharb** [comment]
Let me give it a whirl locally :-)

**Comment by ljharb** [comment]
ok, this technique is awesome!

I think we need a bit more polish though:

I don't think this should happen inside `getRepositories` - I think we should have a shared function that's used in both `src/commands/detail.js` and `src/commands/ls.js`, something like this:

```js
async function loadingIndicator(task) {
  process.stdout.write("Loading...");
  const timer = setInterval(
    () => { process.stdout.write('.'); },
    1e3
  );
  try {
    return await task();
  } finally {
    clearInterval(timer);
    process.stdout.clearLine(0);
    process.stdout.cursorTo(0);
  }
}
```
which is then used like:
```js
const { points, repositories } = await loadingIndicator(() => getRepositories(flags, filter));
```

What do you think?

**Comment by KhatiaIvanova** [comment]
It will work. We can update in that way.

**Comment by KhatiaIvanova** [comment]
Let me give some time @ljharb

**Comment by abdumamdouh** [comment]
Hello @ljharb , can you check now?

File: src/commands/ls.js [hunk 2]
```diff
@@ -9,7 +10,7 @@ module.exports = async function ls(flags) {
 	}
 
 	// Get all repositories
-	const { repositories } = await getRepositories(flags, filter);
+  const { repositories } = await loadingIndicator(() => getRepositories(flags, filter));
```
Comment (ljharb) [inline src/commands/ls.js:13]:
```suggestion
	const { repositories } = await loadingIndicator(() => getRepositories(flags, filter));
```

**Review by ljharb — APPROVED** [review]
Looks pretty good!

**Comment by abdumamdouh** [comment]
> Looks pretty good!

@ljharb  ready to be merged now?

**Comment by ljharb** [comment]
Yep! I'm rebasing it now, and will land it shortly.

**Pull request merged**